src/wp-includes/revision.php

```diff
@@ -463,6 +513,101 @@ function wp_restore_post_revision( $revision, $fields = null ) {
 	return $post_id;
 }
 
+/**
+ * Restore the revisioned meta values for a post.
+ *
+ * @param int $post_id     The ID of the post to restore the meta to.
+ * @param int $revision_id The ID of the revision to restore the meta from.
+ *
+ * @since 6.4.0
+ */
+function wp_restore_post_revision_meta( $post_id, $revision_id ) {
+
+	// Restore revisioned meta fields.
+	foreach ( wp_post_revision_meta_keys( get_post_type( $post_id ) ) as $meta_key ) {
+
+		// Clear any existing meta.
+		delete_post_meta( $post_id, $meta_key );
+
+		_wp_copy_post_meta( $revision_id, $post_id, $meta_key );
+	}
+}
+
+/**
+ * Copy post meta for the given key from one post to another.
+ *
+ * @param int    $source_post_id Post ID to copy meta value(s) from.
+ * @param int    $target_post_id Post ID to copy meta value(s) to.
+ * @param string $meta_key       Meta key to copy.
+ *
+ * @since 6.4.0
+ */
+function _wp_copy_post_meta( $source_post_id, $target_post_id, $meta_key ) {
+
+	foreach ( get_post_meta( $source_post_id, $meta_key ) as $meta_value ) {
```

Comment:
@spacedmonkey - I think the same issue for defaults applies here. If a default is set for a meta field, this code will copy the _default_ value over instead of the actual (missing) value, which is unintentional. switching is to get_metadata_raw as well

Reply:
Changing this to `get_metadata_raw( 'post', $source_post_id, $meta_key = '' )` broke numerous revision group tests so leaving as is.